src/components/sectionList/translation/TranslationDialog.tsx

```diff
@@ -0,0 +1,83 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Modal,
+    ModalContent,
+    ModalTitle,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import React, { useMemo } from 'react'
+import { BaseListModel } from '../../../lib'
+import { Loader } from '../../loading'
+import { TranslationForm } from './TranslationForm'
+import { useDBLocales, useLastSelectedLocale } from './translationFormHooks'
+
+type TranslationDialogProps = {
+    onClose: () => void
+    model: BaseListModel
+}
+
+export const TranslationDialog = ({
+    onClose,
+    model,
+}: TranslationDialogProps) => {
+    const dbLocalesQuery = useDBLocales()
+
+    const [selectedLocaleString, setSelectedLocaleString] =
+        useLastSelectedLocale()
+
+    const selectedLocale = useMemo(() => {
+        if (!selectedLocaleString) {
+            return undefined
+        }
+        return dbLocalesQuery.data?.find(
+            (locale) => locale.locale === selectedLocaleString
+        )
+    }, [selectedLocaleString, dbLocalesQuery.data])
+
+    return (
+        <Modal onClose={onClose} large={true}>
+            <ModalTitle>
+                {i18n.t('Translate {{modelName}}', {
+                    modelName: model.displayName,
+                })}
+            </ModalTitle>
+            <Loader queryResponse={dbLocalesQuery}>
+                <ModalContent>
+                    <SingleSelectField
+                        filterable={true}
+                        label="Locales"
```

Comment:
Needs translation and singular. Eg.
```suggestion
                        label={i18n.t('Locale')}
```